rmw_opendds_cpp/src/process_topic_and_service_names.cpp

```diff
@@ -23,54 +23,22 @@
 #include "rmw_opendds_shared_cpp/namespace_prefix.hpp"
 #include "rmw_opendds_shared_cpp/opendds_include.hpp"
 
-bool
-_process_topic_name(
-  const char * topic_name,
+void
+get_topic_name(
+  const std::string topic_name,
```

Comment:
```suggestion
  const std::string& topic_name,
```

Reply:
Looks like in the final version, `topic_name` became a `const char *`. Why?

Reply:
Because we eliminated the external checks, and saved unnecessary temporary std::string creation on the call stack and in concatenations. Here, we concatenate only when necessary. We still have to create the return std::string. Depending on the compiler setting, we may not have a copy overhead if c++17 (copy elision) is used.